provider/kubernetes/kubernetes.go

```diff
@@ -992,24 +992,25 @@ func getForwardAuthConfig(i *extensionsv1beta1.Ingress, k8sClient Client) (*type
 		AuthResponseHeaders: getSliceStringValue(i.Annotations, annotationKubernetesAuthForwardResponseHeaders),
 	}
 
-	authSecretName := getStringValue(i.Annotations, annotationKubernetesAuthForwardTLSSecret, "")
-	if len(authSecretName) > 0 {
-		authSecretCert, authSecretKey, err := loadAuthTLSSecret(i.Namespace, authSecretName, k8sClient)
-		if err != nil {
-			return nil, fmt.Errorf("failed to load auth secret: %s", err)
-		}
+	authSecretCert, authSecretKey, err := loadAuthTLSSecret(i.Namespace, getStringValue(i.Annotations, annotationKubernetesAuthForwardTLSSecret, ""), k8sClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load auth secret: %s", err)
+	}
 
-		forwardAuth.TLS = &types.ClientTLS{
-			Cert:               authSecretCert,
-			Key:                authSecretKey,
-			InsecureSkipVerify: getBoolValue(i.Annotations, annotationKubernetesAuthForwardTLSInsecure, false),
-		}
+	forwardAuth.TLS = &types.ClientTLS{
```

Comment:
With your algorithm `forwardAuth.TLS` will never be `nil`, it's not the expected behavior:
https://github.com/containous/traefik/blob/4cae8bcb104a515d32c6174687ac0624bb047908/templates/kubernetes.tmpl#L89-L94

So you must do that:
```go

	authSecretName := getStringValue(i.Annotations, annotationKubernetesAuthForwardTLSSecret, "")
	authSecretCert, authSecretKey, err := loadAuthTLSSecret(i.Namespace, authSecretName, k8sClient)
	if err != nil {
		return nil, fmt.Errorf("failed to load auth secret: %s", err)
	}

	if authSecretCert != "" || authSecretKey != "" {
		forwardAuth.TLS = &types.ClientTLS{
			Cert:               authSecretCert,
			Key:                authSecretKey,
			InsecureSkipVerify: getBoolValue(i.Annotations, annotationKubernetesAuthForwardTLSInsecure, false),
		}
	}

	if forwardAuth.TLS == nil && label.Has(i.Annotations, getAnnotationName(i.Annotations, annotationKubernetesAuthForwardTLSInsecure)) {
		forwardAuth.TLS = &types.ClientTLS{
			InsecureSkipVerify: getBoolValue(i.Annotations, annotationKubernetesAuthForwardTLSInsecure, false),
		}
	}

	return forwardAuth, nil
```